### PR TITLE
Add forms

### PR DESCRIPTION
**Pull request by luminitacristinalazarescu**
Issue #24 #40

### CODE REVIEW

File: projects/views.py [hunk 1]
```diff
@@ -1,6 +1,22 @@
 from .models import Project
 
+from django.shortcuts import render
 from django.views.generic.list import ListView
+from django.http import HttpResponseRedirect
+
+from .forms import UploadForm
+
+
+def get_project(request):
```
Comment (mei-li) [inline projects/views.py]: I think you can use a class-based view as well https://docs.djangoproject.com/en/1.8/topics/class-based-views/generic-editing/

File: projects/urls.py [hunk 2]
```diff
@@ -1,9 +1,10 @@
+from projects import views
 from django.conf.urls import url
-
```
Comment (gotche) [inline projects/urls.py:2]: For imports I follow this pattern:

```
standard library imports

third party libraries imports

project imports
```

groups separated by empty line.

I don't know if this project follows any conventions respect this

File: projects/migrations/0003_auto_20151120_2121.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        ('projects', '0002_create_project_model'),
+    ]
+
+    operations = [
+        migrations.AlterField(
+            model_name='project',
+            name='difficulty',
+            field=models.CharField(choices=[('beginner', 'beginner'), ('intermediate', 'intermediate'), ('advanced', 'advanced')], max_length=15),
+        ),
+    ]
```
Comment (gotche) [inline projects/migrations/0003_auto_20151120_2121.py]: this migration is missing the actual change in the models. Usually is a good practice to create migrations (models change + migration file) as a single PR.

File: projects/templates/projects/upload_project.html [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+        <br>
+        {{ form.description.label }}: {{ form.description }}
+        <br>
+        {{ form.max_members.label }}: {{ form.max_members}}
+        <br>
+        {{ form.difficulty.label }}: {{ form.difficulty }}
+        <br>
+        {{ form.owner.label }}: {{ form.owner }}
+        <br>
+        {{ form.participants.label }}: {{ form.participants }}
```
Comment (gotche) [inline projects/templates/projects/upload_project.html]: why not simply `{{form}} and play with the fields in the UploadForm class? It reduces repetition and typos

File: projects/templates/projects/upload_project.html [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+
+{% block content %}
+    <div class="content-section-a">
+
+        <div class="container">
+            <div class="row">
+                <div class="col-lg-5 col-sm-6">
+                    <hr class="section-heading-spacer">
+                    <div class="clearfix"></div>
+                    <form action="/projects/" method="post">
```
Comment (gotche) [inline projects/templates/projects/upload_project.html]: I think this action should point at `/upload/`

**Comment by lluminita** [comment]
@mei-li and @gotche Could you please have another look on it? I fixed the comments and removed the migrations from this PR because I am not sure what happened there, I will fix the migrations in another PR. Thank you.

**Comment by lluminita** [comment]
Fixes https://github.com/lluminita/codeeve/issues/40

**Comment by gotche** [comment]
the last three commits could be squashed together

File: projects/templates/projects/project_list.html [hunk 6]
```diff
@@ -4,11 +4,12 @@
 {% endblock %}
 
 {% block content %}
-{% for project in object_list %}
     <div class="content-section-a">
 
         <div class="container">
             <div class="row">
+                <a href="upload/">Upload a project</a>
```
Comment (mei-li) [inline projects/templates/projects/project_list.html]: Why "upload a project" it is "create project". You can upload a file not a project.

File: projects/urls.py [hunk 7]
```diff
@@ -1,9 +1,9 @@
...
-from projects.views import ProjectListView
+from projects.views import ProjectListView, ProjectCreate
 
 
 urlpatterns = [
 
-    url(r'^$', ProjectListView.as_view(), name='list_projects')
+    url(r'^$', ProjectListView.as_view(), name='project_list'),
+    url(r'^upload/$', ProjectCreate.as_view(), name='project_create'),
```
Comment (mei-li) [inline projects/urls.py]: why not create here as well?

**Comment by mei-li** [comment]
Btw the commit history is very chaotic, i would them reset all and then do the commits again in 2-3 logical units.

File: projects/templates/projects/project_list.html (hunk 6 above)
Comment (lluminita) [inline projects/templates/projects/project_list.html]: Ah, didn't realised :) Will change to create project.

File: projects/urls.py (hunk 7 above)
Comment (lluminita) [inline projects/urls.py]: again, didn't realised it, will change to create.

**Comment by lluminita** [comment]
Ok, after a few issues I reseted everything and created just 1 commit, since there aren't that many changes.

**Comment by mei-li** [comment]
:+1:

**Pull request merged**